### ISSUE TITLE
Don't allow package creation with two versions of the same package in scope

### ISSUE THREAD

**Issue by wardweistra** [FirelyTeam/firely-net-sdk#2416]
**Is your feature request related to a problem? Please describe.**
Packages are being created depending on two different versions of the same package, which leads to multiple different resources with the same canonical being in scope.

**Describe the solution you'd like**
Can we have the package library check on package creation time whether multiple versions of the same package are somewhere in the entire dependency tree and, if so, throw an error?

**Describe alternatives you've considered**
We could also consider to allow this and give the user the option to specify the priority, but this will likely give more problems than it solves at this point.

**Comment by brianpos** [comment]
The canonical resources within these packages should have different canonical versions, and thus should be able to be resolved using a version aware resolver.
I've written a function for performing the version resolution, and "current-canonical" implementaiton too.
I'd love to see this incorporated into the toolchain rather than prevent the situation from occurring.
https://github.com/brianpos/fhir-net-web-api/blob/feature/r4b/src/Hl7.Fhir.WebApi.Support/CurrentCanonical.cs

**Comment by mmsmits** [comment]
I don't think this is doable/practical. Package creators should have the responsibility to check this. 
Also, like @brianpos mentions, when this is the case, you can always specify which version you want in your references.
The FhirPackageSource has this option, and will fallback to the highest version of the canonical resource when no version is specified.
